### PR TITLE
TSK-1142 Removed JUnit4 dependency in taskana-simplehistory-provider

### PR DESCRIPTION
**Pull request by sofie29**
Sonarcloud passed: https://sonarcloud.io/dashboard?branch=TSK-1142&id=sofie29_taskana

### CODE REVIEW

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/SimpleHistoryServiceImpl.java [hunk 1]
```diff
@@ -59,4 +59,13 @@ public void create(TaskanaHistoryEvent event) {
   public HistoryQuery createHistoryQuery() {
     return new HistoryQueryImpl(taskanaHistoryEngine, historyQueryMapper);
   }
+
+  /*
+   * ATTENTION: This method exists for testing purposes.
+   */
+  @VisibleForTesting
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/SimpleHistoryServiceImpl.java]: This was a stupid suggestion - Sorry. Please remove together with the guava dependency. Thanks!

File: history/taskana-simplehistory-provider/pom.xml [hunk 2]
```diff
@@ -64,21 +64,9 @@
...
 		<dependency>
-			<groupId>org.junit.vintage</groupId>
-			<artifactId>junit-vintage-engine</artifactId>
-			<version>${version.junit.jupiter}</version>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>org.powermock</groupId>
-			<artifactId>powermock-api-mockito2</artifactId>
-			<version>${version.powermock}</version>
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/pom.xml]: is this property in use? I don't think so. - Please verify and remove it in the taskana-parent.pom. Thanks!

File: history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java [hunk 3]
```diff
@@ -36,9 +33,9 @@ public void testListValuesAscendingAndDescending() {
             .createHistoryQuery()
             .listValues(HistoryQueryColumnName.COMMENT, SortDirection.DESCENDING);
 
-    assertEquals(3, ascendingList.size());
-    assertArrayEquals(defaultList.toArray(), ascendingList.toArray());
-    assertEquals(ascendingList.get(2), descendingList.get(0));
+    assertThat(ascendingList).hasSize(3);
+    assertThat(ascendingList.toArray()).isEqualTo(defaultList.toArray());
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java]: Please use the assertJ ListAssert.

File: history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java [hunk 4]
```diff
@@ -51,316 +48,315 @@ public void testComplexQuery() {
...
+    assertThat(results.get(0).getUserId()).isEqualTo("admin");
+    assertThat(results.get(1).getUserId()).isEqualTo("peter");
 
     results = query.orderByUserId(SortDirection.DESCENDING).list();
-    assertEquals(2, results.size());
-    assertEquals("admin", results.get(0).getUserId());
-    assertEquals("peter", results.get(1).getUserId());
-    assertEquals(3, query.domainLike().count());
+    assertThat(results).hasSize(2);
+    assertThat(results.get(0).getUserId()).isEqualTo("admin");
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java]: Again, ListAssert :) Thanks!

File: history/taskana-simplehistory-provider/src/test/java/pro/taskana/simplehistory/impl/HistoryQueryImplTest.java [hunk 5]
```diff
@@ -58,7 +58,7 @@ public void testShouldReturnList() throws SQLException {
             .list();
 
     validateMockitoUsage();
-    assertArrayEquals(returnList.toArray(), result.toArray());
+    assertThat(result.toArray()).isEqualTo(returnList.toArray());
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/test/java/pro/taskana/simplehistory/impl/HistoryQueryImplTest.java]: Please use ListAssert

File: history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java (hunk 3 above)
Comment (mustaphazorgati) [reply history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryHistoryAccTest.java]: There is still a `toArray`. Is it necessary? I don't think so.

File: history/taskana-simplehistory-provider/src/test/java/pro/taskana/simplehistory/impl/HistoryQueryImplTest.java (hunk 5 above)
Comment (mustaphazorgati) [reply history/taskana-simplehistory-provider/src/test/java/pro/taskana/simplehistory/impl/HistoryQueryImplTest.java]: There is still a `toArray`. Is it necessary? I don't think so.

**Comment by sofie29** [comment]
> 
> 
> If you want to remove the unnecessary `toArray` calls I'd be happy. But no need to do another roundtrip if you don't want to.

I've already tried to remove it but errors occurred, so I didn't delete them. But I'll look at it again :)

**Pull request merged**